crates/db_views_actor/src/person_mention_view.rs

```diff
@@ -289,26 +296,17 @@ impl<'a> PersonMentionQueryBuilder<'a> {
       query = query.filter(person_mention::read.eq(false));
     }
 
-    query = match self.sort.unwrap_or(SortType::Hot) {
-      SortType::Hot | SortType::Active => query
-        .order_by(hot_rank(comment_aggregates::score, comment_aggregates::published).desc())
+    if !self.show_bot_accounts.unwrap_or(true) {
+      query = query.filter(person::bot_account.eq(false));
+    };
+
+    query = match self.sort.unwrap_or(CommentSortType::Hot) {
+      CommentSortType::Hot => query
+        .then_order_by(hot_rank(comment_aggregates::score, comment_aggregates::published).desc())
         .then_order_by(comment_aggregates::published.desc()),
-      SortType::New | SortType::MostComments | SortType::NewComments => {
-        query.order_by(comment::published.desc())
-      }
-      SortType::TopAll => query.order_by(comment_aggregates::score.desc()),
-      SortType::TopYear => query
-        .filter(comment::published.gt(now - 1.years()))
-        .order_by(comment_aggregates::score.desc()),
-      SortType::TopMonth => query
-        .filter(comment::published.gt(now - 1.months()))
-        .order_by(comment_aggregates::score.desc()),
-      SortType::TopWeek => query
-        .filter(comment::published.gt(now - 1.weeks()))
-        .order_by(comment_aggregates::score.desc()),
-      SortType::TopDay => query
-        .filter(comment::published.gt(now - 1.days()))
-        .order_by(comment_aggregates::score.desc()),
+      CommentSortType::New => query.then_order_by(comment::published.desc()),
+      CommentSortType::Old => query.then_order_by(comment::published.asc()),
+      CommentSortType::Top => query.order_by(comment_aggregates::score.desc()),
```

Comment:
Any particular reason to remove these sort options for comments? Should probably mention this in the changelog too once released.

Reply:
Yep we can do that in the changelog. Post SortType has things like `Active` and `MostComments` which don't make sense for comment sorting, so I made another type.